rskj-core/src/main/java/co/rsk/pcc/bto/GetMultisigScriptHash.java

```diff
@@ -0,0 +1,150 @@
+
+    @Override
+    public Object execute(Object[] arguments) {
+        if (arguments == null || arguments[0] == null) {
+            throw new NativeContractIllegalArgumentException(REQUIRED_SIGNATURE_NULL_OR_ZERO);
+        }
+        int minimumSignatures = ((BigInteger) arguments[0]).intValueExact();
+        Object[] publicKeys = (Object[]) arguments[1];
+
+        if (minimumSignatures <= 0) {
+            throw new NativeContractIllegalArgumentException(REQUIRED_SIGNATURE_NULL_OR_ZERO);
+        }
+
+        if (publicKeys == null || publicKeys.length < MINIMUM_REQUIRED_KEYS) {
+            throw new NativeContractIllegalArgumentException(PUBLIC_KEYS_NULL_OR_ONE);
+        }
+
+        if (publicKeys.length < minimumSignatures) {
+            throw new NativeContractIllegalArgumentException(String.format(
+                    INVALID_REQUIRED_SIGNATURE_AND_PUBLIC_KEYS_PAIR,
+                    publicKeys.length, minimumSignatures
+            ));
+        }
+
+        if (publicKeys.length > MAXIMUM_ALLOWED_KEYS) {
+            throw new NativeContractIllegalArgumentException(String.format(
+                    "Given public keys (%d) are more than the maximum allowed signatures (%d)",
+                    publicKeys.length, MAXIMUM_ALLOWED_KEYS
+            ));
+        }
+
+        List<BtcECKey> btcPublicKeys = new ArrayList<>();
+        Arrays.stream(publicKeys).forEach(o -> {
+            byte[] publicKey = (byte[]) o;
+            if (publicKey.length != COMPRESSED_PUBLIC_KEY_LENGTH && publicKey.length != UNCOMPRESSED_PUBLIC_KEY_LENGTH) {
+                throw new NativeContractIllegalArgumentException(String.format(
+                        "Invalid public key length: %d", publicKey.length
+                ));
+            }
+
+            // Avoid extra work by not recalculating compressed keys on already compressed keys
+            try {
+                BtcECKey btcPublicKey = BtcECKey.fromPublicOnly(publicKey);
+                if (publicKey.length == UNCOMPRESSED_PUBLIC_KEY_LENGTH) {
+                    btcPublicKey = BtcECKey.fromPublicOnly(btcPublicKey.getPubKeyPoint().getEncoded(true));
+                }
+                btcPublicKeys.add(btcPublicKey);
+            } catch (IllegalArgumentException e) {
+                throw new NativeContractIllegalArgumentException(String.format(
+                        "Invalid public key format: %s", Hex.toHexString(publicKey)
+                ), e);
+            }
+        });
+
+        Script multisigScript = ScriptBuilder.createP2SHOutputScript(minimumSignatures, btcPublicKeys);
+
+        return multisigScript.getPubKeyHash();
+    }
+
+    @Override
+    public long getGas(Object[] parsedArguments, byte[] originalData) {
+        int numberOfKeys = ((Object[]) parsedArguments[1]).length;
+
+        // Base cost is the cost for 2 keys (the minimum).
+        // Then a fee is payed per additional key.
+        return BASE_COST + (numberOfKeys - 2) * COST_PER_EXTRA_KEY;
```

Comment:
where is checked that `numberOfKeys` is >= 2 ?

Reply:
https://github.com/rsksmart/rskj/blob/BTO-utils/rskj-core/src/main/java/co/rsk/pcc/bto/GetMultisigScriptHash.java#L86

Reply:
I understand `getGas` is called before `execute` and if a negative value is returned bad things can happen.
https://github.com/rsksmart/rskj/blob/2a7420f35c3786b2c94099717df9404165f2503c/rskj-core/src/main/java/org/ethereum/vm/program/Program.java#L1255-L1263

Reply:
The real danger here is that we don't check that the argument could be null, thus throwing an NPE.
It's true that we didn't do a good job here checking the value before using it in the operation, but given the costs here the gas won't be negative.
Good catch!